ocean-client/src/programs/send-dfi-program.ts

```diff
@@ -0,0 +1,116 @@
+import { IStore } from '../utils/store'
+import { Telegram } from '../utils/telegram'
+import { WalletSetup } from '../utils/wallet-setup'
+import { CommonProgram } from './common-program'
+import { fromAddress } from '@defichain/jellyfish-address'
+import { AccountToUtxos, TransactionSegWit } from '@defichain/jellyfish-transaction/dist'
+import BigNumber from 'bignumber.js'
+
+export class SendDFIProgramm extends CommonProgram {
+  readonly toAddress: string
+  readonly threshold: number
+  readonly sendAll: boolean
+
+  constructor(store: IStore, walletSetup: WalletSetup) {
+    super(store, walletSetup)
+    this.toAddress = this.settings.toAddress
+    this.threshold = this.settings.sendThreshold ?? 1
+    this.sendAll = this.settings.sendAll
+  }
+
+  async doChecks(telegram: Telegram): Promise<boolean> {
+    if (!this.doValidationChecks(telegram, false)) {
+      return false
+    }
+
+    return true
+  }
+
+  async doSend(telegram: Telegram): Promise<boolean> {
+    if (!this.settings.toAddress) {
+      return false
+    }
+    const utxoBalance = await this.getUTXOBalance()
+    console.log('utxo: ' + utxoBalance)
+
+    const balances = await this.getTokenBalances()
+    const dfiBalance = balances.get('DFI')?.amount ?? '0'
+    console.log('dfi: ' + dfiBalance)
+
+    const fromBalance = new BigNumber(dfiBalance)
+
+    let fromUtxos: BigNumber
+    if (this.sendAll) {
+      fromUtxos = utxoBalance
+    } else {
+      fromUtxos = utxoBalance.isGreaterThan(1) ? utxoBalance.minus(1) : new BigNumber(0)
+    }
+    let amountToUse = fromUtxos.plus(fromBalance)
+    console.log('amountToUse: ' + amountToUse)
+
+    if (amountToUse.toNumber() < this.threshold) {
+      console.log('Treshold not reached')
+      await telegram.log('threshold not reached.')
+      return true
+    }
+
+    const account = await this.walletSetup.getAccount(this.getAddress())
+
+    if (fromBalance.toNumber() > 0) {
+      const accountToUtxos: AccountToUtxos = {
+        from: this.script!,
+        balances: [{ token: 0, amount: fromBalance }],
+        mintingOutputsStart: 2,
+      }
+
+      const txn = await account?.withTransactionBuilder()?.account.accountToUtxos(accountToUtxos, this.script!)
+      if (txn === undefined) {
+        await telegram.send('ERROR: transactionSegWit is undefined')
+        console.error('transactionSegWit is undefinedd')
+        return false
+      }
```

Comment:
building tx normally happens in common-program. There you can use the already proven ways to build defiTx. then you also don't need to make this.script protected.